src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java

```diff
@@ -0,0 +1,210 @@
+package com.minecolonies.coremod.commands;
+
+import static com.minecolonies.coremod.commands.ActionArgumentType.STRING;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+
+public class ActionMenuState
+{
+    @NotNull private final ActionMenu actionMenu;
+    @NotNull private final Map<String, ArgumentState> argumentStateByActionArgumentNameMap = new HashMap<>();
+
+    private static class ArgumentState
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a constructor to the class, or provide default values. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1258)